### PR TITLE
MP-2371. Trade any asset

### PR DESCRIPTION
**Pull request by piobab**
We don't support not whitelisted coins. Currently we use whitelisting in Credit Manager contract for:
- deposit
- borrow
- lend
- swap
- zap
- enter vault

This PR removes whitelisting for deposit and swap to be able to trade any asset. Moreover, not whitelisted assets don't count towards Health Factor.

### CODE REVIEW

**Review by markonmars — APPROVED** [review]
Nice! Simple and clean :+1:

**Comment by bobthebuidlr** [comment]
Good solution, especially great use case for `continue`. Overall this is much easier than expected 👍

**Comment by piobab** [comment]
Copied our conclusion regarding HLS:
"Discussed on the call: Don't think we should support this. We should withdraw rewards straight to wallet not to complicate things. Holding those assets in the HLS account doesn't have any added benefit aside from seeing the value increase in the account. We'd still need to add another button to then withdraw the assets to the wallet regardless, as you won't be able to do anything with those assets."

This means also that trade any asset will work only on Default account.

File: contracts/credit-manager/tests/tests/test_deposit.rs [hunk 1]
```diff
@@ -115,42 +113,52 @@ fn deposit_but_not_enough_funds() {
 }
 
 #[test]
-fn can_only_deposit_allowed_assets() {
-    let blacklisted_coin = blacklisted_coin();
-    let not_listed_coin = ujake_info().to_coin(234);
+fn can_deposit_not_allowed_assets() {
```
Comment (bobthebuidlr) [inline contracts/credit-manager/tests/tests/test_deposit.rs:116]:
```suggestion
fn can_deposit_not_whitelisted_assets() {
```

**Review by bobthebuidlr — APPROVED** [review]
Just some naming comments. I see `whitelisted` has `allowed` and `supported` as synonyms. I'm up for just stating `whitelisted`, but do with it what you will:)

File: contracts/credit-manager/tests/tests/test_deposit.rs [hunk 2]
```diff
@@ -115,42 +113,52 @@ fn deposit_but_not_enough_funds() {
...
+    let blacklisted_coin_info = blacklisted_coin();
```
[the remark this answers is not in the document]
Comment (piobab) [reply contracts/credit-manager/tests/tests/test_deposit.rs]: Had to rename more places. See last commit

**Comment by piobab** [comment]
@bobthebuidlr applied naming.

**Pull request merged**